docs/reference/cluster/reroute.asciidoc

```diff
@@ -68,3 +68,5 @@ The commands supported are:
     allocate the shard to. It also accepts `allow_primary` flag to
     explicitly specify that it is allowed to explicitly allocate a primary
     shard (might result in data loss).
+    
+WARNING: Using `allow_primary` in conjunction with `allocate` command to force a primary shard allocation, will cause data contained in existing targeted shard to be lost. Use with caution. 
```

Comment:
How about:

    The `allow_primary` parameter will force a new _empty_primary shard to be allocated, without
    any data. If a node rejoins the cluster later on, and has a copy of the original primary shard,
    including data, that data will be deleted: the old shard copy will be replaced by the new
    live shard copy.

(still not perfect)